File: src/main/java/com/vc/hard/L899.scala
package com.vc.hard

object L899 {
  def orderlyQueue(S: String, K: Int): String = {
    if(K > 1) return S.sorted
    else {
      var sOld = S
      (0 until S.length).foreach(i => {
        val sNew = S.substring(i) + S.substring(0, i)
        sOld = if(sNew > sOld) sOld else sNew
      })
      sOld
    }
  }
}

File: src/main/java/com/vc/hard/L899.java
package com.vc.hard;

import java.util.Arrays;

class L899 {
    public String orderlyQueue(String S, int K) {
        if(K > 1) {
            char[] arr = S.toCharArray();
            Arrays.sort(arr);
            return new String(arr);
        }
        else {
            String res = S;
            String sNew = S;
            for(int i = 0; i < S.length(); i++) {
                //System.out.println(sNew);
                sNew = S.substring(i) + S.substring(0, i);
                if(sNew.compareTo(res) < 0) res = sNew;
            }
            return res;
        }
    }
}
